### PR TITLE
Change geometry outlined building domain

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

File: src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Requests/Building/ChangeBuildingOutlineLambdaRequest.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                sqsRequest.TicketId,
+                sqsRequest.IfMatchHeaderValue,
+                sqsRequest.ProvenanceData.ToProvenance(),
+                sqsRequest.Metadata,
+                sqsRequest.Request)
+        { }
+
+        public ChangeBuildingOutlineLambdaRequest(
+            string messageGroupId,
+            int buildingPersistentLocalId,
```
Comment (jvandaal) [inline src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Requests/Building/ChangeBuildingOutlineLambdaRequest.cs:35]: Maybe passing through the buildingPersistentLocalId is redundant as the messageGroupId should be equal to it?

File: src/BuildingRegistry/Building/Building.cs [hunk 2]
```diff
@@ -50,6 +50,51 @@ public partial class Building : AggregateRootEntity, ISnapshotable
...
+            {
+                throw new BuildingHasBuildingUnitsOutsideBuildingGeometryException();
+            }
+
+            var buildingUnitsWithPositionDerivedFromBuilding = plannedOrRealizedBuildingUnits
+                .Where(x => x.BuildingUnitPosition.GeometryMethod == BuildingUnitPositionGeometryMethod.DerivedFromObject)
+                .Select(x => x.BuildingUnitPersistentLocalId)
+                .ToList();
+
+            var buildingUnitsPosition = plannedOrRealizedBuildingUnits.Any()
```
Comment (ArneD) [inline src/BuildingRegistry/Building/Building.cs:87]: shouldn't this be `buildingUnitsWithPositionDerivedFromBuilding` ? If so, best make a unit test for it.

File: src/BuildingRegistry/Building/Events/BuildingOutlineWasChanged.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        [EventPropertyDescription("Metadata bij het event.")]
+        public ProvenanceData Provenance { get; private set; }
+
+        public BuildingOutlineWasChanged(
+            BuildingPersistentLocalId buildingPersistentLocalId,
+            IEnumerable<BuildingUnitPersistentLocalId> buildingUnitPersistentLocalIds,
+            ExtendedWkbGeometry extendedWkbGeometryBuilding,
+            ExtendedWkbGeometry? extendedWkbGeometryBuildingUnits)
+        {
+            BuildingPersistentLocalId = buildingPersistentLocalId;
```
Comment (ArneD) [inline src/BuildingRegistry/Building/Events/BuildingOutlineWasChanged.cs:39]: So if I'm reading the PR correctly. If `BuildingUnitPersistentLocalIds` is empty, then `ExtendedWkbGeometryBuildingUnits` will be null and vice versa?
Can we add a check in here? If one of the two is null/empty and the other isn't, `throw new ArgumentNullException` ?

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/BuildingRegistry/Building/Events/BuildingOutlineWasChanged.cs (hunk 3 above)
Comment (jvandaal) [reply src/BuildingRegistry/Building/Events/BuildingOutlineWasChanged.cs:39]: That's correct.

File: src/BuildingRegistry/Building/Building.cs (hunk 2 above)
Comment (jvandaal) [reply src/BuildingRegistry/Building/Building.cs:87]: Correct. The variable buildingUnitsWithPositionDerivedFromBuilding should be used. There's already a test covering that the Center of the new building outline is used. I think the test suffices even though the mistake was not catched https://github.com/Informatievlaanderen/building-registry/blob/849293b02e27da1e5aaf17235544aa444fd2a38b/test/BuildingRegistry.Tests/AggregateTests/WhenChangingBuildingOutline/GivenBuildingExists.cs#L62-L108

Comment (jvandaal) [reply src/BuildingRegistry/Building/Building.cs]: Changed to use the correct variable.

Comment (ArneD) [reply src/BuildingRegistry/Building/Building.cs]: I would make a test that does covers this case. I think if you have a case with one buildingunit (planned/realized) not derived, you can already catch this.

File: src/BuildingRegistry/Building/Events/BuildingOutlineWasChanged.cs (hunk 3 above)
Comment (jvandaal) [reply src/BuildingRegistry/Building/Events/BuildingOutlineWasChanged.cs:40]: Added check.

File: src/BuildingRegistry/Building/Building.cs (hunk 2 above)
Comment (jvandaal) [reply src/BuildingRegistry/Building/Building.cs]: Added following test `WithPlannedOrRealizedBuildingUnitsForWhichThePositionIsAppointedByAdministrator_ThenBuildingUnitsPositionRemainsUnchanged`

Now we have:
* ThenBuildingGeometryWasChanged()
* WithSameGeometry_ThenNothing
* WithPlannedOrRealizedBuildingUnitsForWhichThePositionIsDerived_ThenBuildingUnitsPositionWasAlsoChanged()
* WithPlannedOrRealizedBuildingUnitsForWhichThePositionIsAppointedByAdministrator_ThenBuildingUnitsPositionRemainsUnchanged

**Review by ArneD — APPROVED** [review]

**Pull request merged**